src/Field.php

```diff
@@ -532,6 +530,13 @@ public function compare($value, $value2 = null): bool
         return $typecastFunc($value) === $typecastFunc($value2);
     }
 
+    public function getReference(): ?Reference
+    {
+        return $this->referenceLink !== null
+            ? $this->getOwner()->getRef($this->referenceLink)
```

Comment:
I do not think we need to check if `getOwner()` is set, as Field with reference is created always with a model. If not, invoking `getOwner` will throw. :)